flytekit/image_spec/image_spec.py

```diff
@@ -206,6 +210,16 @@ def build(cls, image_spec: ImageSpec):
             click.secho(f"Image {img_name} not found. Building...", fg="blue")
             if image_spec.builder not in cls._REGISTRY:
                 raise Exception(f"Builder {image_spec.builder} is not registered.")
+            if image_spec.builder == "envd":
+                envd_version = metadata.version("envd")
+                # flytekit v1.10.2+ copies the workflow code to the WorkDir specified in the Dockerfile. However, envd<0.3.39
+                # overwrites the WorkDir when building the image, resulting in a permission issue when flytekit downloads the file.
+                if Version(envd_version) < Version("0.3.39"):
```

Comment:
should we also add a lower-bound version in the envd [plugin](https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-envd/setup.py#L7)?

Reply:
I think we don't need to add that. People can upgrade their `flytekitplugins-envd` without upgrading flytekit. if they didn't upgrade flytekit, they don't need to upgrade envd.